frontend/src/views/Collection/index.tsx

```diff
@@ -143,9 +150,13 @@ const Collection: FC = () => {
   const handleOnChange = function (newTabId: TABS) {
     setSelectedTab(newTabId);
   };
+  const hasWriteAccess = collection.access_type === ACCESS_TYPE.WRITE;
+  const shouldShowPrivateWriteAction = hasWriteAccess && isPrivate;
 
-  const shouldShowPrivateWriteAction =
-    collection.access_type === ACCESS_TYPE.WRITE && isPrivate;
+  const handleDeleteCollection = () => {
+    deleteMutation({ collectionID: id, visibility: VISIBILITY_TYPE.PUBLIC });
```

Comment:
maybe await here before redirect, so the user doesn't see the deleted collection in My Collections while delete is in flight?

```ts
await deleteMutation(...);
```

Reply:
Nice catch. Even better, let's optimistically remove the collection!

713ba0e

Reply:
Ahh great idea 💡 !

I think we still need to await `deleteMutation()` at line 157 😆